File: services/administration-service/src/main/java/sgptt/adminsvc/entity/Person.java
package sgptt.adminsvc.entity;

import java.util.Date;

import jakarta.persistence.*;
import lombok.*;

@Data
@RequiredArgsConstructor
@AllArgsConstructor
@NoArgsConstructor

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public abstract class Person {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    @Column(name="person_id")
    private Long personId;

    @NonNull
    @Column(name="name", nullable = false, length = 50)
    private String name;

    @NonNull
    @Column(name="paternal_surname", nullable = false, length = 50)
    private String paternalSurname;

    @NonNull
    @Column(name="maternal_surname", nullable = false, length = 50)
    private String maternalSurname;

    @NonNull
    @Column(name="email", nullable = false, length = 70, unique = true)
    private String email;

    @NonNull
    @Column(name="password", nullable = false, length = 100)
    private String password;

    @NonNull
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name="created_at")
    private Date createdAt;

    @Column(name="is_active", nullable = false, columnDefinition = "BOOLEAN DEFAULT TRUE")
    private Boolean isActive = true;
}

File: services/protocols-service/src/main/kotlin/com/sgptt/protocolsservice/repository/entity/Person.kt
package com.sgptt.protocolsservice.repository.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Inheritance
import jakarta.persistence.InheritanceType
import jakarta.persistence.Temporal
import jakarta.persistence.TemporalType
import java.util.Date

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
abstract class Person(
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "person_id", nullable = false)
	val personId: Long,
	
	@Column(name = "name", nullable = false, length = 50)
	val name: String,
	
	@Column(name = "paternal_surname", nullable = false, length = 50)
	val paternalSurname: String,
	
	@Column(name = "maternal_surname", nullable = false, length = 50)
	val maternalSurname: String,
	
	@Column(name = "email", nullable = false, length = 70, unique = true)
	val email: String,
	
	@Column(name = "password", nullable = false, length = 100)
	val password: String,
	
	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "created_at", nullable = false)
	val createdAt: Date,
	
	@Column(name = "is_active", nullable = false, columnDefinition = "BOOLEAN DEFAULT TRUE")
	val isActive: Boolean
	
)